### ISSUE TITLE
TypeError at /login/

### ISSUE THREAD

**Issue by merit-mthompson** [jazzband/django-debug-toolbar#1317]
Django Debug Toolbar version: 2.2

With the following view:

```python
from django.contrib.auth.views import LoginView

class PublicLoginView(LoginView):
    redirect_field_name = None
```

Getting the following error:

```bash
Environment:

Request Method: GET
Request URL: http://(removed)/login/

Django Version: 2.2.2
Python Version: 3.6.9
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.humanize',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.forms',
 'capture_tag',
 'django_celery_beat',
 'django_celery_results',
 'debug_toolbar']
Installed Middleware:
['debug_toolbar.middleware.DebugToolbarMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'global_login_required.GlobalLoginRequiredMiddleware']

Traceback:

File "/home/django/srv/current/venv/lib/python3.6/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/home/django/srv/current/venv/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
  145.                 response = self.process_exception_by_middleware(e, request)

File "/home/django/srv/current/venv/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
  143.                 response = response.render()

File "/home/django/srv/current/venv/lib/python3.6/site-packages/django/template/response.py" in render
  106.             self.content = self.rendered_content

File "/home/django/srv/current/venv/lib/python3.6/site-packages/django/template/response.py" in rendered_content
  83.         content = template.render(context, self._request)

File "/home/django/srv/current/venv/lib/python3.6/site-packages/django/template/backends/django.py" in render
  61.             return self.template.render(context)

File "/home/django/srv/current/venv/lib/python3.6/site-packages/django/template/base.py" in render
  171.                     return self._render(context)

File "/home/django/srv/current/venv/lib/python3.6/site-packages/django/test/utils.py" in instrumented_test_render
  95.     template_rendered.send(sender=self, template=self, context=context)

File "/home/django/srv/current/venv/lib/python3.6/site-packages/django/dispatch/dispatcher.py" in send
  175.             for receiver in self._live_receivers(sender)

File "/home/django/srv/current/venv/lib/python3.6/site-packages/django/dispatch/dispatcher.py" in <listcomp>
  175.             for receiver in self._live_receivers(sender)

File "/home/django/srv/current/venv/lib/python3.6/site-packages/debug_toolbar/panels/templates/panel.py" in _store_template_info
  91.                 key_values = sorted(context_layer.items())

Exception Type: TypeError at /login/
Exception Value: '<' not supported between instances of 'NoneType' and 'str'
```

This only happens when `redirect_field_name` is set to `None`. If I remove it or change it to a string things work as expected and there aren't any errors.

**Comment by tim-schilling** [comment]
The issue here is that `LoginView` has a key-value pair in it's context with the `redirect_field_name` as a key. You can override the `get_context_data` method in `PublicLoginView` or find an alternate way to achieve your goal.

```
class PublicLoginView(LoginView):
    redirect_field_name = None

        def get_context_data(self, **kwargs):
            context = super().get_context_data(**kwargs)
            if self.redirect_field_name is None:
                context.pop(self.redirect_field_name, None)
            return context
```

Why are you setting `redirect_field_name` to `None`? Maybe we can find a different way to solve that for you.

**Comment by merit-mthompson** [comment]
@tim-schilling thanks, we'll look at modifying the context. What we've done for now is set redirect_field_name to an unused/absurd string to keep DDT from throwing errors due to the `sorted()` use and setting `redirect_field_name` to `None`.

The use case for removing the redirect field is to help curb some auth-related redirect loops for certain views by disabling the "next" feature of the login view.

**Comment by jdufresne** [comment]
IMO, this should be considered incorrect use of `LoginView` and therefore closed. Template variable names are expected to be strings, so using `None` would be API misuse. 

@merit-mthompson one option would be to override get_redirect_url to always return `''`.

https://github.com/django/django/blob/f1f24539d8c86f60d1e2951a19eb3178e15d6399/django/contrib/auth/views.py#L69-L80

**Comment by merit-mthompson** [comment]
Django has a test specifically for `None` which leads me to believe it's at least allowed: https://github.com/django/django/blob/cc7c16af98e4c9a7146c248c8567aac7a50e0fbb/tests/auth_tests/test_mixins.py#L156

Feel free to close, we're working around it. The fix would be to change how things are being looped/sorted in DDT at https://github.com/jazzband/django-debug-toolbar/blob/master/debug_toolbar/panels/templates/panel.py#L91. Maybe pop None items out before sorting.

**Comment by jdufresne** [comment]
Thanks for linking to that test. I agree we should support this if Django does.

I think we could sort by `repr()` as a workaround. Are you interested in submitting a PR with a test?

**Comment by jdufresne** [comment]
Potential fix in #1385

**Comment by merit-mthompson** [comment]
@jdufresne thank you!
